### PR TITLE
Add bf16 matmul, fc, elementwise add and mul

### PR DESCRIPTION
**Pull request by wozna**
### PR types

New features
### PR changes

 OPs
### Describe

This PR presents:
- bfloat16 kernels for matmul, fc, elementwise_add and elementwise_mul 
- unit tests for all introduced operators
- adds check of `use_mkldnn` attribute

### CODE REVIEW

File: paddle/fluid/framework/ir/graph_pattern_detector.cc [hunk 1]
```diff
@@ -2101,13 +2101,18 @@ PDNode *patterns::QuantizePlacement::operator()(
...
+          {"concat", "conv2d", "elementwise_add", "elementwise_mul", "fc",
+           "fusion_gru", "gelu", "layer_norm", "matmul", "reshape2", "softmax",
+           "sum", "transpose2"});
   if (!bfloat16_enabled_op_types.empty()) {
     supported_op_types = bfloat16_enabled_op_types;
   }
   auto *op = pattern->NewNode(op_repr())->assert_is_ops(supported_op_types);
+  op->assert_more([&](Node *node) {
+    return node->Op()->GetAttrIfExists<bool>("use_mkldnn") ||
+           node->Op()->Type() == "reshape2";
```
Comment (jczaja) [inline paddle/fluid/framework/ir/graph_pattern_detector.cc:2114]: why reshape  in this pattern does not have to be mkldnn==True ?

Comment (jczaja) [reply paddle/fluid/framework/ir/graph_pattern_detector.cc:2114]: ok, now it is clear.

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by wozna** [comment]
@luotao1  Could you please start your review? Your approval will be needed for PR-CI-APROVAL

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**